src/coreclr/vm/siginfo.cpp

```diff
@@ -386,6 +386,21 @@ void SigPointer::ConvertToInternalSignature(Module* pSigModule, SigTypeContext *
         cArgs--;
     }
 }
+
+void SigPointer::CopySignature(Module* pSigModule, SigBuilder* pSigBuilder, SigPointer* pSigPtrTokenEnd, BYTE additionalCallConv)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END
+
+    BYTE element;
+    IfFailThrowBF(GetByte(&element), BFA_BAD_COMPLUS_SIG, pSigModule);
+    pSigBuilder->AppendByte(element | additionalCallConv);
+    pSigBuilder->AppendBlob((const PVOID)m_ptr, pSigPtrTokenEnd->m_ptr - m_ptr);
```

Comment:
```suggestion
    SigPointer spEnd(*this);
    IfFailThrowBF(spEnd.SkipSignature(), BFA_BAD_COMPLUS_SIG, pSigModule);

    BYTE element;
    IfFailThrowBF(PeekByte(&element), BFA_BAD_COMPLUS_SIG, pSigModule);
    pSigBuilder->AppendByte(element | additionalCallConv);
    pSigBuilder->AppendBlob((const PVOID)m_ptr, spEnd.m_ptr - m_ptr);
```
If you would like to keep this as separate method, I would move the logic that computes the length of the signature into this method too, so that it is more self-contained.

Reply:
Keeping as a separate method avoids leaking the protected ptr and length fields.

I removed the incoming "end" sigpointer as shown above. Also since "GetByte()" is no longer called, m_ptr must have + 1 added.

Also, for the method to no have unexpected side-effects, it either needs to advance m_ptr all the way to the end or just leave it as-is at the beginning. I just left it as-is and did the +1 and -1 for the calling convention byte.